### PR TITLE
revive #600

### PR DESCRIPTION
**Pull request by eristhea**

### CODE REVIEW

**Comment by eristhea** [comment]
all comments are addressed from the original PR. we can consider this to already have my approval

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/block/entity/BannerBlockEntity.mapping [hunk 1]
```diff
@@ -3,6 +3,7 @@ CLASS net/minecraft/unmapped/C_mwamopji net/minecraft/block/entity/BannerBlockEn
 	FIELD f_ejwtyvwp PATTERNS_KEY Ljava/lang/String;
 	FIELD f_rkjwxvms baseColor Lnet/minecraft/unmapped/C_arllgqae;
 	FIELD f_ufvgulvv customName Lnet/minecraft/unmapped/C_rdaqiwdt;
+	FIELD f_yqkmzniy defaultPatternComponent Lnet/minecraft/unmapped/C_wwbphpbp;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/block/entity/BannerBlockEntity.mapping]:
```suggestion
	FIELD f_yqkmzniy patterns Lnet/minecraft/unmapped/C_wwbphpbp;
```
it's just initialized to the default, it's reassigned from data elsewhere

File: mappings/net/minecraft/block/entity/BannerBlockEntity.mapping [hunk 2]
```diff
@@ -11,4 +12,6 @@ CLASS net/minecraft/unmapped/C_mwamopji net/minecraft/block/entity/BannerBlockEn
 		ARG 2 state
 		ARG 3 baseColor
 	METHOD m_bxugddgd getPickStack ()Lnet/minecraft/unmapped/C_sddaxwyk;
+	METHOD m_igihlswz (Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_arllgqae;)V
+		ARG 2 dyeColor
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/block/entity/BannerBlockEntity.mapping]:
```suggestion
		ARG 2 baseColor
```
more specific, and matches the field it's assigned to

File: mappings/net/minecraft/enchantment/Enchantments.mapping [hunk 3]
```diff
@@ -1 +1,4 @@
 CLASS net/minecraft/unmapped/C_lkwxngnl net/minecraft/enchantment/Enchantments
+	METHOD m_fsqdbhms registerEnchantment (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/Enchantments.mapping]:
```suggestion
	METHOD m_fsqdbhms createRegistryKey (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
```
it just creates the key, registration happens in `bootstrap`
In #607 I'll make this consistent for everything with a similar method and a `bootstrap` method called from `VanillaDynamicRegistries`.

**Comment by supersaiyansubtlety** [comment]
re-reviewing because I found some issues when updating #607

File: mappings/net/minecraft/enchantment/Enchantments.mapping [hunk 4]
```diff
@@ -1 +1,4 @@
...
+	METHOD m_fzzlzdym bootstrap (Lnet/minecraft/unmapped/C_hqoyyfco;)V
+	METHOD m_zklinuaf registerEnchantment (Lnet/minecraft/unmapped/C_hqoyyfco;Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_jxtrubuh$C_epmoqirc;)V
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/enchantment/Enchantments.mapping]:
```suggestion
	METHOD m_zklinuaf register (Lnet/minecraft/unmapped/C_hqoyyfco;Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_jxtrubuh$C_epmoqirc;)V
```
I think the params and containing class make it clear that this registers an enchantment

File: mappings/net/minecraft/block/dispenser/ProjectileDispenserBehavior.mapping [hunk 5]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/unmapped/C_bcbkqaaq net/minecraft/block/dispenser/ProjectileDispenserBehavior
 	FIELD f_lnvxbzxz projectile Lnet/minecraft/unmapped/C_dokwmwgh;
+	FIELD f_uupspuna dispenserConfig Lnet/minecraft/unmapped/C_dokwmwgh$C_jscrvllj;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/block/dispenser/ProjectileDispenserBehavior.mapping]:
```suggestion
	FIELD f_uupspuna config Lnet/minecraft/unmapped/C_dokwmwgh$C_jscrvllj;
```
I think the containing class makes it clear

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/enchantment/Enchantments.mapping (hunk 3 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/enchantment/Enchantments.mapping]: actually how about
```suggestion
	METHOD m_fsqdbhms createKey (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
```
or
```suggestion
	METHOD m_fsqdbhms keyOf (Ljava/lang/String;)Lnet/minecraft/unmapped/C_xhhleach;
```
(less conventional, but I think it works? maybe we could update conventions?)

**Comment by eristhea** [comment]
thanks for the suggestions, i think i'll go with `createKey` for now and we can discuss reevaluating the name

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**